### PR TITLE
Search for artists from dialog

### PR DESCRIPTION
**Pull request by rahul-raturi**
Allow searching for artists from a dialog. I'm not sure whether it will be super useful, but it's better for consistency with other two dialog which are *Album* (PR: https://github.com/metabrainz/picard/pull/486) and *Track* (PR: https://github.com/metabrainz/picard/pull/475). This dialog displays basic information about the artist, and provides an option to look him/her up in web browser.

### Screenshot
screenshot from 2016-08-20 15-55-42

### Note
This PR is based on rebase of #486. It would be helpful if review upto commit 851b581 be provided there.

### CODE REVIEW

**Comment by zas** [comment]
@rahul-raturi : can you rebase this branch ?

**Comment by rahul-raturi** [comment]
@zas, Yes, working on it.

**Comment by rahul-raturi** [comment]
Rebased.

File: picard/ui/searchdialog.py [hunk 1]
```diff
@@ -715,3 +722,102 @@ def save_state(self):
...
+        ]
+
+    def search(self, text):
+        self.retry_params = (self.search, text)
+        self.search_box.search_edit.setText(text)
+        self.show_progress()
+        self.tagger.xmlws.find_artists(self.handle_reply,
+                query=text,
+                search=True,
+                limit=25)
```
Comment (zas) [inline picard/ui/searchdialog.py]: Use `QUERY_LIMIT` here

**Comment by zas** [comment]
This PR looks fine to me.

I think we should implement paging in those dialogs, so the user can retrieved next 25 entries etc...
We can add `offset` keyword param to https://github.com/rahul-raturi/picard/blob/8d2934c063be3eeaa7032c6b663b836fed417a5b/picard/webservice.py#L457 (in another PR).

**Comment by zas** [comment]
@rahul-raturi : i also noticed an annoying thing, search of mbids doesn't work anymore !

If you paste a text (or url) with mbid in Search field, it should either open the matching entity page in browser, or load the release in Picard (release mbids).
Related code doesn't seem to be executed anymore.
See 
https://github.com/metabrainz/picard/blob/09df7f8fb73aa25f7d06b1892fea26a923a8cced/picard/browser/filelookup.py#L79 and following lines.

Can you fix this issue please ?

File: picard/ui/searchdialog.py (hunk 1 above)
Comment (rahul-raturi) [inline picard/ui/searchdialog.py]: Done, squashed into 939f44a.

**Comment by rahul-raturi** [comment]
> I think we should implement paging in those dialogs, so the user can retrieved next 25 entries etc...

I'd discussed this with @mwiencek once. What I came to know is search results tend to be mostly irrelevant below this limit. The user may rather change the search query. But if it's necessary to show more results, we can either:
1. Provide predefined limits from a combobox.
2. Paging as you suggested. (I haven't checked whether it's possible with qt).

**Pull request merged**